ast-nodes/type-declaration.dart

```diff
@@ -41,6 +41,6 @@ class TypeDeclaration extends Declaration {
   }
 
   void checkSemantics() {
-    // TODO: implement
+    value.checkSemantics();
```

Comment:
Also ensure there's no other declaration with the same name in the scope

Reply:
Oh, get it. IsReserved checks only program key words, not names..